app/schemas.py

```diff
@@ -1,9 +1,61 @@
+import re
+from datetime import date
+from datetime import datetime
+
 from pydantic import BaseModel
 from pydantic import ConfigDict
+from pydantic import Field
+from pydantic import field_serializer
+from pydantic import field_validator
+from sqlalchemy_utils import Currency
+
+from app.enums import PriceLocationOSMType
 
 
 class UserBase(BaseModel):
     model_config = ConfigDict(from_attributes=True)
 
     user_id: str
     token: str
+
+
+class PriceCreate(BaseModel):
+    model_config = ConfigDict(from_attributes=True, arbitrary_types_allowed=True)
+
+    product_code: str = Field(min_length=1)
+    price: float
+    currency: str | Currency
```

Comment:
why `str | Currency`? Don't you want to coerce values to a fixed set of currency?

Reply:
I went with https://sqlalchemy-utils.readthedocs.io/en/latest/data_types.html#module-sqlalchemy_utils.types.currency

Reply:
allows `currency.code`, `currency.name` & `currency.symbol`
but a bit of serialization as well to avoid errors